pkg/controller/registry/resolver/operators.go

```diff
@@ -331,6 +342,10 @@ func NewOperatorFromBundle(bundle *api.Bundle, startingCSV string, sourceKey Cat
 		}
 	}
 
+	if len(depList) < 1 {
```

Comment:
You can append to a nil slice, so instead of doing this it could be:
```
var depList []VersionDependency
...
depList = append(depList, vd)
```
and depList will be `nil` if `append` is never called.